clients/githubrepo/graphql.go

```diff
@@ -205,19 +212,42 @@ func (handler *graphqlHandler) setup() error {
 			"issueCommentsToAnalyze": githubv4.Int(issueCommentsToAnalyze),
 			"reviewsToAnalyze":       githubv4.Int(reviewsToAnalyze),
 			"labelsToAnalyze":        githubv4.Int(labelsToAnalyze),
-			"commitsToAnalyze":       githubv4.Int(commitsToAnalyze),
+			"commitsToAnalyze":       githubv4.Int(CommitsToAnalyze),
 			"commitExpression":       githubv4.String(commitExpression),
+			"historyCursor":          (*githubv4.String)(nil),
 		}
-		if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
-			handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
+		if CommitsToAnalyze < 99 {
+			if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
+				handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
+				return
+			}
+			handler.archived = bool(handler.data.Repository.IsArchived)
+			handler.issues = issuesFrom(handler.data)
+			handler.commits, handler.errSetup = commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
 			return
 		}
-		handler.archived = bool(handler.data.Repository.IsArchived)
-		handler.commits, handler.errSetup = commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
-		if handler.errSetup != nil {
-			return
+		var allCommits []clients.Commit
+		commitsLeft := CommitsToAnalyze
+		vars["commitsToAnalyze"] = githubv4.Int(100)
+		for ; commitsLeft > 0; commitsLeft = commitsLeft - 100 {
```

Comment:
I would put these in a dedicated function to simplify the code here.

Reply:
would something like this make sense ? having two sepererate functions one for paging and one not?
```
func (handler *graphqlHandler) setupWithPaging() {
	commitExpression := handler.commitExpression()
	vars := map[string]interface{}{
		"owner":                  githubv4.String(handler.repourl.owner),
		"name":                   githubv4.String(handler.repourl.repo),
		"pullRequestsToAnalyze":  githubv4.Int(pullRequestsToAnalyze),
		"issuesToAnalyze":        githubv4.Int(issuesToAnalyze),
		"issueCommentsToAnalyze": githubv4.Int(issueCommentsToAnalyze),
		"reviewsToAnalyze":       githubv4.Int(reviewsToAnalyze),
		"labelsToAnalyze":        githubv4.Int(labelsToAnalyze),
		"commitsToAnalyze":       githubv4.Int(CommitsToAnalyze),
		"commitExpression":       githubv4.String(commitExpression),
		"historyCursor":          (*githubv4.String)(nil),
	}
	var allCommits []clients.Commit
	commitsLeft := CommitsToAnalyze
	for vars["commitsToAnalyze"] = githubv4.Int(100); commitsLeft > 0; commitsLeft = commitsLeft - 100 {
		if commitsLeft < 100 {
			vars["commitsToAnalyze"] = githubv4.Int(commitsLeft)
		}
		if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
			handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
			break
		}
		vars["historyCursor"] = handler.data.Repository.Object.Commit.History.PageInfo.EndCursor
		tmp, err := commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
		if err != nil {
			handler.errSetup = err
			break
		}
		allCommits = append(allCommits, tmp...)
	}
	handler.commits = append(handler.commits, allCommits...)
	handler.issues = issuesFrom(handler.data)
	handler.archived = bool(handler.data.Repository.IsArchived)
}


func (handler *graphqlHandler) setupDefault() {
	commitExpression := handler.commitExpression()
	vars := map[string]interface{}{
		"owner":                  githubv4.String(handler.repourl.owner),
		"name":                   githubv4.String(handler.repourl.repo),
		"pullRequestsToAnalyze":  githubv4.Int(pullRequestsToAnalyze),
		"issuesToAnalyze":        githubv4.Int(issuesToAnalyze),
		"issueCommentsToAnalyze": githubv4.Int(issueCommentsToAnalyze),
		"reviewsToAnalyze":       githubv4.Int(reviewsToAnalyze),
		"labelsToAnalyze":        githubv4.Int(labelsToAnalyze),
		"commitsToAnalyze":       githubv4.Int(CommitsToAnalyze),
		"commitExpression":       githubv4.String(commitExpression),
		"historyCursor":          (*githubv4.String)(nil),
	}
	if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
		handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
		return
	}
	handler.archived = bool(handler.data.Repository.IsArchived)
	handler.issues = issuesFrom(handler.data)
	handler.commits, handler.errSetup = commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
	return
}


func (handler *graphqlHandler) setup() error {
	// if --number-of-commits flag set to greater than 99 we need to page to query the github api.
	// else run default setup no paging required.
	if CommitsToAnalyze > 99 {
		handler.setupOnce.Do(func() {
			handler.setupWithPaging()
		})
	} else {
		handler.setupOnce.Do(func() {
			handler.setupDefault()
		})
	}
	return handler.errSetup
}
```

Reply:
I'd personally share the code that is shared, and only create a function `populateCommits()` for:
```
for vars["commitsToAnalyze"] = githubv4.Int(100); commitsLeft > 0; commitsLeft = commitsLeft - 100 {
		if commitsLeft < 100 {
			vars["commitsToAnalyze"] = githubv4.Int(commitsLeft)
		}
		if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
			handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
			break
		}
		vars["historyCursor"] = handler.data.Repository.Object.Commit.History.PageInfo.EndCursor
		tmp, err := commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
		if err != nil {
			handler.errSetup = err
			break
		}
		allCommits = append(allCommits, tmp...)
	}
```

Wdut?

Reply:
yes that makes sense, something like this ?


```
func populateCommits(handler *graphqlHandler, vars map[string]interface{}) ([]clients.Commit, error) {
	var allCommits []clients.Commit
	commitsLeft := CommitsToAnalyze
	for vars["commitsToAnalyze"] = githubv4.Int(100); commitsLeft > 0; commitsLeft = commitsLeft - 100 {
		if commitsLeft < 100 {
			vars["commitsToAnalyze"] = githubv4.Int(commitsLeft)
		}
		err := handler.client.Query(handler.ctx, handler.data, vars)
		if err != nil {
			return nil, fmt.Errorf("failed to populate commits: %w", err)
		}
		vars["historyCursor"] = handler.data.Repository.Object.Commit.History.PageInfo.EndCursor
		tmp, err := commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
		if err != nil {
			return nil, fmt.Errorf("failed to populate commits: %w", err)
		}
		allCommits = append(allCommits, tmp...)
	}
	return allCommits, nil
}```

Reply:
yes, thanks